### PR TITLE
Password about to expire warning.

### LINKED ISSUE #18: Password about to expire warning
There is a setting for this purpose `PASSWORD_EXPIRY_WARNING_DAYS`, but no code to support it.

Add code that can be called to find out when the users should be warned, and how many days they have left to change their password.

Also add a signal for the password is about to expire so end user can send emails etc.

### PR DESCRIPTION
**Pull request by sztamas**
Fixes #18

### CODE REVIEW

File: useraudit/password_expiry.py [hunk 1]
```diff
@@ -154,12 +154,23 @@ def set_password_change_date(user, date):
...
 
 
-def is_password_expired(user):
+def should_warn_about_password_expiry(user):
+    warn_in_days = ExpirySettings.get().num_warning_days or 0
+    days_left = days_to_password_expiry(user) or 0
+    return warn_in_days > 0 and days_left <= warn_in_days
+
+
+def days_to_password_expiry(user):
```
Comment (aahunter) [inline useraudit/password_expiry.py]: This method doesn't always return a value.

**Comment by id2359** [comment]
with:  def is_account_expired(user):
    earliest = ExpirySettings.get().earliest_possible_login
    if earliest:
        last_login = get_user_last_login(user)
        return last_login and last_login < earliest
    return False

that means accounts which have never logged in can't expire, right?

File: useraudit/password_expiry.py (hunk 1 above)
Comment (id2359) [inline useraudit/password_expiry.py]: maybe it's used like  a Maybe datatype: None or Some date

Comment (sztamas) [inline useraudit/password_expiry.py]: yeah, I actually named it `maybe_warn_about_password_expiry` initially :)

**Comment by sztamas** [comment]
Yes.
But I don't think I touched that code.

**Pull request merged**